plugins/usbdmx/ShowJockeyDMXU1.cpp

```diff
@@ -0,0 +1,362 @@
+  if (!handle) {
+    return -1;
+  }
+
+  if (size <= max_packet_size) {
+    int transferred = -1;
+    m_adaptor->BulkTransfer(handle, endpoint, buffer, size,
+                            &transferred, URB_TIMEOUT_MS);
+
+    return transferred;
+  }
+
+  return -1;
+}
+
+// SynchronousShowJockeyDMXU1
+// -----------------------------------------------------------------------------
+
+SynchronousShowJockeyDMXU1::SynchronousShowJockeyDMXU1(
+    LibUsbAdaptor *adaptor,
+    libusb_device *usb_device,
+    const std::string &serial)
+    : ShowJockeyDMXU1(adaptor, usb_device, serial) {
+}
+
+bool SynchronousShowJockeyDMXU1::Init() {
+  libusb_device_handle *usb_handle;
+
+  int interface_number;
+  int endpoint;
+  int max_packet_size;
+  if (!LocateInterface(m_adaptor,
+                       m_usb_device,
+                       &interface_number,
+                       &endpoint,
+                       &max_packet_size)) {
+    return false;
+  }
+
+  bool ok = m_adaptor->OpenDeviceAndClaimInterface(m_usb_device,
+                                                   interface_number,
+                                                   &usb_handle);
+  if (!ok) {
+    return false;
+  }
+
+  std::auto_ptr<ShowJockeyDMXU1ThreadedSender> sender(
+      new ShowJockeyDMXU1ThreadedSender(m_adaptor,
+                                        m_usb_device,
+                                        usb_handle,
+                                        max_packet_size,
+                                        endpoint));
+  if (!sender->Start()) {
+    return false;
+  }
+
+  m_sender.reset(sender.release());
+  return true;
+}
+
+bool SynchronousShowJockeyDMXU1::SendDMX(const DmxBuffer &buffer) {
+  return m_sender.get() ? m_sender->SendDMX(buffer) : false;
+}
+
+// ShowJockeyDMXU1AsyncUsbSender
+// -----------------------------------------------------------------------------
+class ShowJockeyDMXU1AsyncUsbSender : public AsyncUsbSender {
+ public:
+  ShowJockeyDMXU1AsyncUsbSender(LibUsbAdaptor *adaptor,
+                                libusb_device *usb_device,
+                                int endpoint,
+                                int max_packet_size_out,
+                                libusb_device_handle *handle)
+                                : AsyncUsbSender(adaptor, usb_device),
+                                  m_endpoint(endpoint),
+                                  m_max_packet_size_out(max_packet_size_out) {
+    m_usb_handle = handle;
+    m_tx_frame = NULL;
+  }
+
+  ~ShowJockeyDMXU1AsyncUsbSender() {
+    CancelTransfer();
+    delete[] m_tx_frame;
+  }
+
+  libusb_device_handle* SetupHandle() {
+    return m_usb_handle;
+  }
+
+  bool PerformTransfer(const DmxBuffer &buffer) {
+    uint16_t nb_sequence = DMX_MAX_SLOT_NUMBER / (m_max_packet_size_out - 2);
+    nb_sequence += 1;
+    int final_size = DMX_MAX_SLOT_NUMBER + (2 * nb_sequence);
+    if (m_tx_frame == NULL) {
+      m_tx_frame = new uint8_t[final_size]();
```

Comment:
Same here, this can be allocated once in the constructor and reused since we know the max size this can possibly be.